src/Spryker/Zed/AppPayment/Persistence/AppPaymentRepository.php

```diff
@@ -104,6 +118,23 @@ public function getRefundsByTransactionIdAndOrderItemIdAndStatuses(
             ->mapPaymentRefundEntityCollectionToPaymentRefundTransfers($paymentRefundEntityCollection);
     }
 
+    protected function applyPaymentCriteria(
+        SpyPaymentQuery $spyPaymentQuery,
+        PaymentCriteriaTransfer $paymentCriteriaTransfer
+    ): SpyPaymentQuery {
+        $paymentConditionsTransfer = $paymentCriteriaTransfer->getPaymentConditionsOrFail();
+
+        if ($paymentConditionsTransfer->getTenantIdentifier() !== null) {
+            $spyPaymentQuery->filterByTenantIdentifier($paymentConditionsTransfer->getTenantIdentifier());
+        }
```

Comment:
When no tenant identifier set we will delete all payments no matter from who those are. 

Shouldn't we require the tenantIdentifier to be set to prevent deletion of other tenants data?

Reply:
It's required
https://github.com/spryker/app-payment/blob/master/src/Spryker/Zed/AppPayment/Persistence/AppPaymentEntityManager.php#L89

Reply:
You cannot use repository method to delete anything, there's no such code in here.